File: app/src/main/java/com/kfix/sample/SimpleApplication.java
package com.kfix.sample;

import android.app.Application;
import android.content.Context;
import com.kfix.sdk.KFixRuntime;
import com.kfix.sdk.Patch;
import java.io.File;

public class SimpleApplication extends Application {
    @Override
    protected void attachBaseContext(Context base) {
        super.attachBaseContext(base);
        Thread.setDefaultUncaughtExceptionHandler((t, e) -> {
            new KFixRuntime().reset();
            e.printStackTrace();
            SimplePatchManager.removePatch(base);
            System.exit(1);
        });
        String patchPath = SimplePatchManager.getPatch(base);
        if (patchPath != null) {
            File optimizedDirectory = new File(new File(patchPath).getParentFile(), "oDex");
            KFixRuntime kFixRuntime = new KFixRuntime();
            kFixRuntime.apply(this, new Patch(patchPath, optimizedDirectory.getAbsolutePath(), null));
            SimplePatchManager.removePatch(base);
        }
    }
}

File: app/src/main/java/com/kfix/sample/SimpleApplication.kt
package com.kfix.sample

import android.app.Application
import android.content.Context
import com.kfix.sdk.KFixRuntime
import com.kfix.sdk.Patch
import java.io.File
import kotlin.system.exitProcess

class SimpleApplication: Application() {
    override fun attachBaseContext(base: Context) {
        super.attachBaseContext(base)
        Thread.setDefaultUncaughtExceptionHandler { t, e ->
            KFixRuntime().reset()
            SimplePatchManager.removePatch(base)
            exitProcess(1)
        }
        SimplePatchManager.getPatch(base)?.let { path ->
            val optimizedDirectory = File(cacheDir, File(path).nameWithoutExtension)
            optimizedDirectory.deleteRecursively()
            KFixRuntime().apply(
                Patch(
                    path = path,
                    oDexPath = optimizedDirectory.absolutePath,
                    libraryPath = null
                )
            )
            SimplePatchManager.removePatch(base)
        }
    }
}